iOS/testbed/__main__.py

```diff
@@ -36,6 +39,46 @@ class MySystemExit(Exception):
     pass
 
 
+class SimulatorLock:
+    # An fcntl-based filesystem lock that can be used to ensure that
+    def __init__(self, timeout):
+        self.filename = Path(tempfile.gettempdir()) / "python-ios-testbed"
```

Comment:
Does the lock need to be system-wide rather than just for a particular user? If so, the use of `tempfile.gettempdir()` can be problematic here. As documented, it checks first for a `TMPDIR` environment variable and it appears that usually on macOS `TMPDIR` is set to a user-specific temp directory, for example:
```
>>> tempfile.gettempdir()
'/var/folders/sn/0m4rnbyj2z1byjs68sz838300000gn/T'
```

Reply:
As best as I can make out, the lock only needs to be per *user*. If I start a simulator on one user's account, then use Fast User Switching to move to a second user's account, the simulator on the first user's account doesn't show on the second.